File: src/main/java/org/example/FileUtils.java
package org.example;

import java.io.*;
import java.net.URL;

public class FileUtils {

    public static String readResourceFile(String name) {
        URL resource = ClassLoader.getSystemResource(name);
        File file = new File(resource.getPath());
        return convertFileToString(file.getAbsolutePath());
    }

    public static String convertFileToString(String filePath) {
        try {
            FileInputStream fileInputStream = new FileInputStream(filePath);
            BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(fileInputStream));
            StringBuilder stringBuilder = new StringBuilder();
            String line;
            while ((line = bufferedReader.readLine()) != null) {
                stringBuilder.append(line);
                stringBuilder.append("\n"); // 可根据需要自定义换行符
            }
            bufferedReader.close();
            fileInputStream.close();
            return stringBuilder.toString();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

    }
}

File: src/main/kotlin/info/laht/threekt/utils/FileUtils.kt
package info.laht.threekt.utils

import java.io.*

object FileUtils {
    fun readResourceFile(name: String?): String {
        val resource = ClassLoader.getSystemResource(name)
        val file = File(resource.path)
        return convertFileToString(file.absolutePath)
    }

    fun convertFileToString(filePath: String?): String {
        return try {
            val fileInputStream = FileInputStream(filePath)
            val bufferedReader = BufferedReader(InputStreamReader(fileInputStream))
            val stringBuilder = StringBuilder()
            var line: String?
            while (bufferedReader.readLine().also { line = it } != null) {
                stringBuilder.append(line)
                stringBuilder.append("\n") // 可根据需要自定义换行符
            }
            bufferedReader.close()
            fileInputStream.close()
            stringBuilder.toString()
        } catch (e: IOException) {
            throw RuntimeException(e)
        }
    }
}